plugins/modules/cloud/misc/proxmox_nic.py

```diff
@@ -303,10 +310,7 @@ def main():
 
     # If vmid is not defined then retrieve its value from the vm name,
     if not vmid:
-        try:
-            vmid = get_vmid(proxmox, name)[0]
-        except Exception:
-            module.fail_json(msg='VM with name = %s does not exist in cluster' % name)
+        vmid = get_vmid(module, proxmox, name)[0]
```

Comment:
I believe the original try block here was meant to really catch an `IndexError` when the resulting list was empty though it was also catching any generic `ResourceExcpetion` from `proxmoxer` as well. Since you are now ensuring `get_vmid` only returns a single VM it probably makes sense to just return the single VM if it exists or `None` and then trigger failure based on the null check.

Reply:
Thanks, `get_vmid` is now more robust and will only return the vmid when only one result exists. Fixed in the latest commit.